addons/mail/static/src/messaging/entity/message/message.js

```diff
@@ -26,6 +26,31 @@ function MessageFactory({ Entity }) {
             threadCache.update({ checkedMessages: [['link', threadCache.messages]] });
         }
 
+        /**
+         * @static
+         */
+        static computeHasSomeoneFetched() {
+            this.env.entities.Message.all().forEach(message => {
+                message.update({
+                    hasEveryoneFetched: message._computeHasEveryoneFetched(),
+                    hasSomeoneFetched: message._computeHasSomeoneFetched(),
+                    partnersThatHaveFetched: message._computePartnersThatHaveFetched(),
+                });
+            });
+        }
+        static computeSeenValues() {
+            this.env.entities.Message.all().forEach(message => {
+                message.update({
+                    hasEveryoneSeen: message._computeHasEveryoneSeen(),
+                    hasSomeoneFetched: message._computeHasSomeoneFetched(),
+                    hasSomeoneSeen: message._computeHasSomeoneSeen(),
+                    isPreviousToLastCurrentPartnerMessageSeenByEveryone: message._computeIsPreviousToLastCurrentPartnerMessageSeenByEveryone(),
+                    partnersThatHaveFetched: message._computePartnersThatHaveFetched(),
+                    partnersThatHaveSeen: message._computePartnersThatHaveSeen(),
+                });
+            });
+        }
```

Comment:
It was a solution that @seb-odoo gave to me because in some case computed fields were not recomputed (I don't remember the condition for it to happen). In my case, when channel_fetched and channel_seen notification are handled, entities are updated but computed fields are not so I need to force the recompute

Reply:
The idea was to have a working code first, I didn't want you to be stuck with this issue.

But if we have a proper suggestion to fix it now that it works and that we have tests to ensure it works, then by all means it could be changed.

I'm not sure if we have time for that though.

Reply:
Just wanted to point out that this approach should be avoided if possible, because this increases our technical debt. If we need this because there is no way to do the clean way now, this increases considerably the importance of making changes to fixes these workarounds ASAP.

I'm aware time is precious right now, but at the same time I would hate doing fast and dirty implementations and then regret that we could have taken a bit more time to do it better. This workaround adds a lot of (artificially private) compute methods/fields, and I think this adds lots of obstacles in fixing this problem in the imminent future, further slowing down functional bug fixing and technical improvements.

If you can't avoid this, at the very least put a comment on all fields and compute methods that are fake?

Reply:
I don't understand why you say compute methods are fake as they aren't for me ... ? It's just that in case of an update they are not called correctly. For me, with that implementation, when we managed to get it to work we just have to remove these two functions and their calls and all will work fine. Am I wrong ?